public/app/plugins/panel/stat/StatPanel.tsx

```diff
@@ -82,6 +83,8 @@ export class StatPanel extends PureComponent<PanelProps<PanelOptions>> {
 
   getValues = (): FieldDisplay[] => {
     const { data, options, replaceVariables, fieldConfig, timeZone } = this.props;
+    // Grab any custom units to prepend/append
+    const customConfig = fieldConfig.defaults?.custom ?? {};
```

Comment:
this is not a safe way to check if field configs are applied... it only checks if the defaults are set, and does not check if the individual fields are set.

On top of that... the options ui will fill in default values for all of them anyway, so it will always be true 🤯

Reply:
Nice catch; I've updated the logic accordingly.